tst/regression/test_suites/advection_convergence/advection_convergence.py

```diff
@@ -0,0 +1,183 @@
+                'output1/variable=cons',
+                'output1/dt=0.4',
+                'time/tlim=0.4',
+                'mesh/nx1=400']
+        """
+
+        n_res = len(lin_res)
+
+        # TEST: Advection only in x-direction 
+        # using nx2 = nx3 = 4 > 1 for identical errors between dimensions
+        if step <= n_res:
+            parameters.driver_cmd_line_args = [
+                'parthenon/mesh/nx1=%d' % lin_res[step % n_res -1],
+                'parthenon/meshblock/nx1=%d' % lin_res[step % n_res -1],
+                'parthenon/mesh/nx2=4',
+                'parthenon/meshblock/nx2=4',
+                'parthenon/mesh/nx3=4',
+                'parthenon/meshblock/nx3=4',
+                'Advection/vy=0.0',
+                'Advection/vz=0.0',
+                ]
+        # TEST: Advection only in y-direction
+        elif step <= 2*n_res:
+            parameters.driver_cmd_line_args = [
+                'parthenon/mesh/nx1=4',
+                'parthenon/meshblock/nx1=4',
+                'parthenon/mesh/nx2=%d' % lin_res[step % n_res -1],
+                'parthenon/meshblock/nx2=%d' % lin_res[step % n_res -1],
+                'parthenon/mesh/nx3=4',
+                'parthenon/meshblock/nx3=4',
+                'Advection/vx=0.0',
+                'Advection/vz=0.0',
+                ]
+        # TEST: Advection only in z-direction
+        elif step <= 3*n_res:
+            parameters.driver_cmd_line_args = [
+                'parthenon/mesh/nx1=4',
+                'parthenon/meshblock/nx1=4',
+                'parthenon/mesh/nx2=4',
+                'parthenon/meshblock/nx2=4',
+                'parthenon/mesh/nx3=%d' % lin_res[step % n_res -1],
+                'parthenon/meshblock/nx3=%d' % lin_res[step % n_res -1],
+                'Advection/vx=0.0',
+                'Advection/vy=0.0',
+                ]
+        # TEST: Advection at along diagonal with dx != dy != dz and Lx != Ly != Lz (half res)
+        elif step == 16:
+            parameters.driver_cmd_line_args = [
+                'parthenon/mesh/nx1=32',
+                'parthenon/meshblock/nx1=32',
+                'parthenon/mesh/nx2=32',
+                'parthenon/meshblock/nx2=32',
+                'parthenon/mesh/nx3=32',
+                'parthenon/meshblock/nx3=32',
+                ]
+        # TEST: Advection at along diagonal with dx != dy != dz and Lx != Ly != Lz (def res)
+        elif step == 17:
+            parameters.driver_cmd_line_args = [
+                ]
+        # TEST: Advection at along diagonal with dx != dy != dz and Lx != Ly != Lz (def res)
+        # using multiple MeshBlocks
+        elif step == 18:
+            parameters.driver_cmd_line_args = [
+                'parthenon/meshblock/nx1=8',
+                'parthenon/meshblock/nx2=16',
+                'parthenon/meshblock/nx3=8',
+                ]
+
+        return parameters
+
+    def Analyse(self,parameters):
```

Comment:
Maybe we could add plotting code in here and generate a convergence plot.

Reply:
I'm still undecided whether this is sth we want or not as the convergence test should mainly be a regression test of the "integrated framework" and not a the specific advection algorithm.
So from may point of view, knowing that it failed and where (the regression test prints the lines that are off in addition to the dat file as artifact) are sufficient in terms of what this test should achieve.
However, if we the test should be an example of how to produce integrated convergence plots, then I'm happy to add it.

Reply:
> However, if we the test should be an example of how to produce integrated convergence plots, then I'm happy to add it.

I do think this would be useful.

> So from may point of view, knowing that it failed and where (the regression test prints the lines that are off in addition to the dat file as artifact) are sufficient in terms of what this test should achieve.

Even from this point of view, I find it much easier to catch errors and understand what's going on with a plot. The tabulated output is not the easiest to read, whereas a plot will instantly tell us what is going wrong and how.

Reply:
I guess you decided not to add the plot? Fair enough. It's not a huge deal either way.

Reply:
I'll add it now. I wanted to make everything else works first.